selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/site/LoginExistedUserWithOpenShiftOAuthTest.java

```diff
@@ -0,0 +1,164 @@
+ * <br>
+ * <b>Test environment</b>: <br>
+ * Eclipse Che Multiuser deployed on OCP with support of OpenShift OAuth server.<br>
+ * If you are going to deploy Eclipse Che on OCP with {@code deploy/openshift/ocp.sh} script, use
+ * {@code --setup-ocp-oauth} parameter to setup OCP OAuth identity provider.<br>
+ * <br>
+ * <b>Test case</b>:<br>
+ * - register new Eclipse Che user;<br>
+ * - go to login page of Eclipse Che;<br>
+ * - click on button to login with OpenShift OAuth;<br>
+ * - login to OCP with registered earlier Eclipse Che test user credentials;<br>
+ * - authorize ocp-client to access OpenShift account;<br>
+ * - fill first broker profile page;<br>
+ * - add OCP user to existed Che user account;<br>
+ * - login into Eclipse Che again;<br>
+ * - create and open workspace of java type;<br>
+ * - switch to the Eclipse Che IDE and wait until workspace is ready to use;<br>
+ * - go to OCP and check there if there is a project with name equals to test workspace id;<br>
+ * - remove test workspace from Eclipse Che Dashboard;<br>
+ * - go to OCP and check there if there is no project with name equals to test workspace id.<br>
+ * <br>
+ * <a href="https://github.com/eclipse/che/issues/8178">Feature reference.</a> <br>
+ * <br>
+ *
+ * @author Dmytro Nochevnov
+ */
+@Test(groups = {TestGroup.OPENSHIFT, TestGroup.MULTIUSER})
+public class LoginExistedUserWithOpenShiftOAuthTest {
+
+  private static final String WORKSPACE_NAME = NameGenerator.generate("workspace", 4);
+
+  public static final String LOGIN_TO_CHE_WITH_OPENSHIFT_OAUTH_MESSAGE_TEMPLATE =
+      "Authenticate as %s to link your account with openshift-v3";
+
+  public static final String USER_ALREADY_EXISTS_ERROR_MESSAGE_TEMPLATE =
+      "User with email %s already exists. How do you want to continue?";
+
+  @Inject private CheLoginPage cheLoginPage;
+  @Inject private OpenShiftLoginPage openShiftLoginPage;
+  @Inject private FirstBrokerProfilePage firstBrokerProfilePage;
+  @Inject private AuthorizeOpenShiftAccessPage authorizeOpenShiftAccessPage;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private Dashboard dashboard;
+  @Inject private Workspaces workspaces;
+  @Inject private NewWorkspace newWorkspace;
+  @Inject private TestWorkspaceServiceClient defaultUserWorkspaceServiceClient;
+  @Inject private ToastLoader toastLoader;
+  @Inject private Ide ide;
+  @Inject private SeleniumWebDriverHelper seleniumWebDriverHelper;
+  @Inject private OpenShiftProjectCatalogPage openShiftProjectCatalogPage;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private TestDashboardUrlProvider testDashboardUrlProvider;
+
+  @AfterClass
+  private void removeTestWorkspace() throws Exception {
+    defaultUserWorkspaceServiceClient.delete(WORKSPACE_NAME, defaultTestUser.getName());
+  }
+
+  @Test
+  public void checkExistedCheUserOcpProjectCreationAndRemoval() throws Exception {
+    // go to login page of Eclipse Che
```

Comment:
I think this is some of the extra clarifications to write comments after each method.

Reply:
Disagree. I would better to read comment which explain what the step is commonly doing than trying to understand it from list of methods which are called during the step execution.

Reply:
```
    // authorize ocp-client to access OpenShift account
    authorizeOpenShiftAccessPage.allowPermissions();
```
```
    // fill first broker profile page
    firstBrokerProfilePage.submit(defaultTestUser);
```
it looks like 
```
//click on login button
loginForm.clickLoginButton
```

Reply:
I see you point.